Algo.h

```diff
@@ -294,7 +385,8 @@ class MMP
                 // invert wale pains???
 
                 for (auto &e : ((Vertex *)(source->p))->edges) {
-                    Interval ii(0, 0, 0, e->length(), 0, NULL, e);
+                    //check for invert
+                    Interval ii(0, 0, 0, e->length(), 0, NULL, e, false);
```

Comment:
Not really an edge case. If we propagate to those egdes directly, there will be no windows formved (ever) on the edges adjacent to te source vertex. And if the destination point is on one of those edges, our algo won't terminate.

Plus this is how the original implementation does it.